scripts/js/commands/api/generateHistoricalRedirects.ts

```diff
@@ -0,0 +1,101 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { readdir, writeFile } from "fs/promises";
+import { basename, join } from "path";
+
+import { Pkg } from "../../lib/api/Pkg.js";
+import { zxMain } from "../../lib/zx.js";
+import { removeSuffix } from "../../lib/stringUtils.js";
+
+const OUTPUT_FILE = "./scripts/config/historical-pages-to-latest.json";
+
+zxMain(async () => {
+  await generateHistoricalRedirects();
+});
+
+export async function generateHistoricalRedirects(): Promise<void> {
+  console.log(`Generating ${OUTPUT_FILE}`);
+  const redirectData: HistoricalRedirectData = {};
+  for (const packageName of Pkg.VALID_NAMES) {
+    redirectData[packageName] = await getRedirectsForPackage(
+      join("docs/api", packageName),
+    );
+  }
+  await writeFile(OUTPUT_FILE, JSON.stringify(redirectData, null, 2));
```

Comment:
This will make it so Prettier doesn't complain

```suggestion
  await writeFile(OUTPUT_FILE, JSON.stringify(redirectData, null, 2) + "\n");
```

I'm not sure if Prettier will want to rewrite this line for being too long